tech spec for monorepo POC (GSI-659) (#107)
ghga-de/epic-docs: 51-ethiopian-wolf/technical_specification.md

  and releases.
    - We can start by evaluating whether a simple setup in which all services are
      versioned and releases together is sufficient.
- How to manage dependencies for individual services?
    - One pyproject.toml and lock file per service?
    - One global lock file for keeping dependencies shared between services in sync?
    - One docker container per service?
- Overall directory structure? Clarifying where following files and directories are
  located:
    - src dirs for individual services
    - pyproject.tomls
    - example_config.yaml / config_schema.json
    - openapi.yaml
    - Readme(s) (per service or one global?)
    - scripts (e.g. update template script)
    - .devcontainer
    - python tooling settings (linters, formatters, etc.)
    - tests (unit, service-wise API testing, inter-service integration testing, etc.)
- Which types of tests are required in the mono-repo?
    - unit tests (per service)
    - service-wise API tests (conformance tests)
    - inter-service integration tests
        - Behavioral
        - contract-testing
        - black and/or white box
- How to ensure that microservice boundraries are respected?
    - import analysis?

>     - flag PRs that add changes to multiple repositories

Reviewer: ```suggestion
    - flag PRs that add changes to multiple services within the monorepo
```

    - discussion on
      [merging the git history](https://softwareengineering.stackexchange.com/questions/379228/steps-to-convert-multi-repo-to-mono-repo#:~:text=By%20default%20this%20will%20add%20the%20complete%20history%2C,updates%20from%20the%20original%20repo%20%28git%20subtree%20pull%29.)
      of multiple repos into one monorepo
    - https://medium.com/lgtm/migrating-to-the-monorepo-582106142654
- a [vscode-specific monorepo guide](https://github.com/microsoft/vscode-python/wiki/Mono-Repo-Set%E2%80%90up-Guide)
- an generally informative [website on monorepo tooling](https://monorepo.tools/)
- general articles:
    - https://medium.com/opendoor-labs/our-python-monorepo-d34028f2b6fa
    - https://medium.com/pinterest-engineering/building-a-python-monorepo-for-fast-reliable-development-be763781f67
    - https://medium.com/@davidsmithtech/dry-and-efficient-python-mono-repos-with-code-example-bc7ee8292e9d
    - https://betterprogramming.pub/the-pros-and-cons-monorepos-explained-f86c998392e1

## Additional Implementation Details:

Moving the existing file services into a monorepo might serve as a good proof of
concept as these services are fully evolved and understood.

> Comming up with a strategy for life cycle management (versioning, releasing, branching)

Reviewer: ```suggestion
Coming up with a strategy for life cycle management (versioning, releasing, branching)
```

> has highest priority and should servce a long-lasting solution. The other aspects are

Reviewer: ```suggestion
has highest priority and should service a long-lasting solution. The other aspects are
```
